setup.py

```diff
@@ -20,6 +20,13 @@
             ['resource/' + package_name]),
     ],
     install_requires=install_requires,
+    extras_require={
```

Comment:
I'm not against this, but in many of our other python packages we specify:

```
tests_require=['pytest'],
```

Any particular reason for the difference?

Reply:
Sigh.  That is going to be another round of deprecations we are probably going to have to fix for Ubuntu 24.04 (with whatever newer setuptools is in there).

But that's not your fault or the fault of this PR :).  Approving.